types/interval_compound_type.go

```diff
@@ -0,0 +1,68 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+// IntervalCompoundType this is used to represent a type of interval compound.
+type IntervalCompoundType struct {
+	precision        TimePrecision
+	typeVariationRef uint32
+	nullability      Nullability
+}
+
+// NewIntervalCompoundType creates a type of new interval compound.
+func NewIntervalCompoundType() IntervalCompoundType {
+	return IntervalCompoundType{}
+}
+
```

Comment:
should the `typeVariationRef` be a parameter? Or do we need a `WithTypeVariationRef` method?

Reply:
added `WithTypeVariationRef`